Hide table and show message when a sample has 0 files

JeffreyThiessen opened the pull request:
## What does this PR do and why?
When a sample has no files, shows a message instead of rendering an empty table.
Fixes #152

## Screenshots or screen recordings
Sample with an attachment
![image](https://github.com/phac-nml/irida-next/assets/16961365/14a7a4c6-5a59-4379-9175-6501d09fa842)

Sample without any attachments
![image](https://github.com/phac-nml/irida-next/assets/16961365/94f1c925-b029-49ab-b66d-41ffc5d34c11)

## How to set up and validate locally
1. pull branch, run, run tailwind to update ui
2. create a new project
3. create 2 samples
4. add an attachment to one of the samples
5. verify that the appropriate table or message is rendered depending on if the sample has attachments or not.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

JeffreyThiessen commented:
When there are attachments on a sample, and they are all deleted, the table stays on page empty. Refreshing shows the message as intended.

Should we be redrawing the cards on sample delete for this edge case?

joshsadam commented:
Try a more centred layout.  A good reference you might want to take a look at is https://tailwindui.com/components/application-ui/feedback/empty-states#component-42930c785190e14c24b53ba822f4a92c

joshsadam commented:
Can you update the screenshots for this PR?

JeffreyThiessen commented:
> Can you update the screenshots for this PR?
Updated


> Can you add a test that ensures the empty state is displayed when there are no attachments?
I have added asserts to the existing test `'user with role >= Maintainer should be able to attach a file to a Sample'` which checks if the text/files are showing or not in the `#attachments` element before and after adding the file.

ericenns commented on app/assets/stylesheets/application.tailwind.css:
```diff
@@ -465,3 +465,11 @@ body {
   @apply text-red-900 placeholder-red-300 border-red-700;
 }
+
+.table-container table:not(:has(td)) {
```
To solve this, just change the last added class to as follows:

```css
.table-container:not(has(td)) .empty_state_message {
  display: block;
}
```

and add the following:

```css
.table-container .empty_state_message {
  display: none;
}
```

JeffreyThiessen replied:
Updated

joshsadam replied:
Thanks

joshsadam reviewed (approved):
This looks good, thanks @JeffreyThiessen
